source/lac/petsc_parallel_sparse_matrix.cc

```diff
@@ -890,6 +890,42 @@ namespace PETScWrappers
       return u*tmp;
     }
 
+    IndexSet
+    SparseMatrix::locally_owned_domain_indices () const
+    {
+      PetscInt n_rows, n_cols, min, max;
+      PetscErrorCode ierr;
+      IS *rows = nullptr;
+      IS *cols = nullptr;
+
+      ierr = MatGetSize (matrix, &n_rows, &n_cols);
+      ierr = MatGetOwnershipIS(matrix, rows, cols);
```

Comment:
i don't know details, but is PETSc index set object always contiguous for all kinds of matrices? From their documentation there are a lot of different methods to create those IS objects.
On another hand, AFAIK, in matrices range is always contiguous.

It would be good to add an Assert that the IS is indeed contiguous so that `locally_owned_domain_indices.add_range(min, max);` makes sense in future when/if PETSc will start supporting non-contiguous ranges.